Fix travis timeout in BufferHashGrouperTest

jihoonson opened the pull request:
Reduced the buffer size by half. Hopefully this fixes https://github.com/apache/incubator-druid/issues/6712.

jihoonson commented:
From https://github.com/apache/incubator-druid/pull/4333#discussion_r124079021, the buffer size in these tests have some special meaning. I changed this PR to keep `testGrowing3` but use a smaller buffer for `testGrowing2` which still tests the same bug.

clintropolis commented on processing/src/test/java/org/apache/druid/query/groupby/epinephelinae/BufferHashGrouperTest.java:
```diff
@@ -140,8 +140,8 @@ public void testGrowing()
     final TestColumnSelectorFactory columnSelectorFactory = GrouperTestUtil.newColumnSelectorFactory();
-    final Grouper<Integer> grouper = makeGrouper(columnSelectorFactory, 2_000_000_000, 2);
-    final int expectedMaxSize = NullHandling.replaceWithDefault() ? 40988516 : 39141224;
+    final Grouper<Integer> grouper = makeGrouper(columnSelectorFactory, 250_000_000, 2);
```
Should you add a comment to explain why this value is significant? Also if you're feeling ambitious, maybe would be nice to rename tests to something like `testGrowingOverflow` since that is what I think this is testing, and maybe `testGrowingMax` for `testGrowing3`?

jihoonson commented:
It looks that `testGrowing3` still makes the test fail, so I removed it. Also, the `bufferSize` of the previous change wasn't good to test the bug case reported in #4333, so I adjusted it and `maxLoadFactor` again. You can verify that `testGrowingOverflowingInteger()` still checks the bug of #4333 by adding the below code to `ByteBufferHashTable.adjustTableWhenFull()`. It's working if it prints `different !!!!!!!!`.

```
    final boolean a = ((long) maxBuckets * 3 * bucketSizeWithHash + tableStart) > (long) tableArenaSize;
    final boolean b = (maxBuckets * 3 * bucketSizeWithHash + tableStart) > tableArenaSize;
    if (a != b) {
      System.err.println("different !!!!!!!!");
    } else {
      System.out.println("same");
    }
```

jihoonson commented:
I'm trying to figure out why the build gets stuck. Added `WIP` tag.

b-slim commented on .travis.yml:
```diff
@@ -47,13 +47,15 @@ matrix:
         - sh -c "dmesg | egrep -i '(oom|out of memory|kill process|killed).*' -C 1 || exit 0"
         - free -m
+      after_script:
+        - cat processing/target/surefire-reports/org.apache.druid.query.groupby.epinephelinae.BufferHashGrouperTest-output.txt
```
is this for debugging ?
